code/modules/antagonists/eldritch_cult/eldritch_book.dm

```diff
@@ -0,0 +1,158 @@
+	///how many charges do we have?
+	var/charge = 1
+	///Where we cannot create the rune?
+	var/static/list/blacklisted_turfs = typecacheof(list(/turf/closed,/turf/open/space,/turf/open/lava))
+
+/obj/item/forbidden_book/Destroy()
+	last_user = null
+	. = ..()
+
+
+/obj/item/forbidden_book/examine(mob/user)
+	. = ..()
+	if(!user.mind.has_antag_datum(/datum/antagonist/heretic))
+		return
+	. += "The Tome holds [charge] charges."
+	. += "Use it on the floor to create a transmutation rune, used to perform rituals."
+	. += "Hit an influence in the black part with it to gain a charge."
+	. += "Hit a transmutation rune to destroy it."
+
+/obj/item/forbidden_book/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!proximity_flag)
+		return
+	if(istype(target,/obj/effect/eldritch))
+		remove_rune(target,user)
+	if(istype(target,/obj/effect/reality_smash))
+		get_power_from_influence(target,user)
+	if(istype(target,/turf/open))
+		draw_rune(target,user)
+
+///Gives you a charge and destroys a corresponding influence
+/obj/item/forbidden_book/proc/get_power_from_influence(atom/target, mob/user)
+	var/obj/effect/reality_smash/RS = target
+	to_chat(target, "<span class='warning'>You start drawing power from influence...</span>")
+	if(do_after(user,10 SECONDS,FALSE,RS))
+		qdel(RS)
+		charge += 1
+
+///Draws a rune on a selected turf
+/obj/item/forbidden_book/proc/draw_rune(atom/target,mob/user)
+
+	if(!user.mind.has_antag_datum(/datum/antagonist/heretic))
+		return
+	for(var/turf/T in range(1,target))
+		if(is_type_in_typecache(T, blacklisted_turfs))
+			to_chat(target, "<span class='warning'>The terrain doesn't support runes!</span>")
+			return
+	var/A = get_turf(target)
+	to_chat(target, "<span class='warning'>You start drawing a rune...</span>")
+
+	if(do_after(user,30 SECONDS,A))
+
+		new /obj/effect/eldritch/big(A)
+
+///Removes runes from the selected turf
+/obj/item/forbidden_book/proc/remove_rune(atom/target,mob/user)
+
+	if(!user.mind.has_antag_datum(/datum/antagonist/heretic))
+		return
+	to_chat(target, "<span class='warning'>You start removing a rune...</span>")
+	if(do_after(user,2 SECONDS,user))
+		qdel(target)
+
+/obj/item/forbidden_book/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state) // Remember to use the appropriate state.
+	if(!user.mind.has_antag_datum(/datum/antagonist/heretic))
+		return FALSE
+	last_user = user
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "ForbiddenLore", name, 500, 900, master_ui, state)
+		ui.open()
+
+/obj/item/forbidden_book/ui_data(mob/user)
+	var/datum/antagonist/heretic/cultie = user.mind.has_antag_datum(/datum/antagonist/heretic)
+	var/list/to_know_alpha = list()
+	for(var/Y in cultie.get_researchable_knowledge())
+		to_know_alpha += new Y
+	var/list/to_know = get_custom_sorted_list(user,to_know_alpha)
```

Comment:
i have no idea how to sort something JS side

Reply:
and i want a very special type of sorting, that is based on the path, so that main paths are always at the top, while side paths show up at the bottom